test/general/fork.pass.cpp

```diff
@@ -0,0 +1,13 @@
+#include <oneapi/dpl/execution>
+#include "support/utils.h"
```

Comment:
The inclusion of this header causes this new test to fail because the test utility headers define a static global policy:
https://github.com/oneapi-src/oneDPL/blob/e3a4ed39d5306dad1582a33eb7ac85035a26dbab/test/support/utils_sycl.h#L100

Once I removed this include (and changed the last line in `main` to `return 0`), this test now passes with the Codeplay NVIDIA backend.

Reply:
After looking into it further, I believe the test header `utils_sycl.h` might be incorrect. It is creating a queue based on the macro `ONEDPL_USE_PREDEFINED_POLICIES`; however, I believe at that point it should be using `_ONEDPL_PREDEFINED_POLICIES` instead.

https://github.com/oneapi-src/oneDPL/blob/e3a4ed39d5306dad1582a33eb7ac85035a26dbab/test/support/utils_sycl.h#L99-L103

If I change the test header to use the latter macro, then your test passes without modification.

Reply:
Thanks Adam, I will take a look.
As for the macro, preferably tests should use public APIs (including macros) and avoid internal ones. Maybe this is the place to do it differently, but I'd need to make sure.

Reply:
@adamfidel You were right, the definition in `utils_sycl.h` did not properly handle the "default" behavior for `ONEDPL_USE_PREDEFINED_POLICIES` which is as if it was defined to 1 (and not 0, the preprocessor default value). I have updated that file.